src/nupic/regions/SDRClassifierRegion.py

```diff
@@ -0,0 +1,500 @@
+          count=0,
+          regionLevel=True,
+          isDefaultOutput=False,
+          requireSplitterMap=False),
+      ),
+
+      parameters=dict(
+        learningMode=dict(
+          description='Boolean (0/1) indicating whether or not a region '
+                      'is in learning mode.',
+          dataType='UInt32',
+          count=1,
+          constraints='bool',
+          defaultValue=1,
+          accessMode='ReadWrite'),
+
+        inferenceMode=dict(
+          description='Boolean (0/1) indicating whether or not a region '
+                      'is in inference mode.',
+          dataType='UInt32',
+          count=1,
+          constraints='bool',
+          defaultValue=0,
+          accessMode='ReadWrite'),
+
+        maxCategoryCount=dict(
+          description='The maximal number of categories the '
+                      'classifier will distinguish between.',
+          dataType='UInt32',
+          required=True,
+          count=1,
+          constraints='',
+          # arbitrarily large value for backward compatibility 
+          defaultValue=1000,
+          accessMode='Create'),
+
+        steps=dict(
+          description='Comma separated list of the desired steps of '
+                      'prediction that the classifier should learn',
+          dataType="Byte",
+          count=0,
+          constraints='',
+          defaultValue='0',
+          accessMode='Create'),
+
+        alpha=dict(
+          description='The alpha used to compute running averages of the '
+                      'bucket duty cycles for each activation pattern bit. A '
+                      'lower alpha results in longer term memory',
+          dataType="Real32",
+          count=1,
+          constraints='',
+          defaultValue=0.001,
+          accessMode='Create'),
+
+        implementation=dict(
+          description='The classifier implementation to use.',
+          accessMode='ReadWrite',
+          dataType='Byte',
+          count=0,
+          constraints='enum: py, cpp'),
+
+        clVerbosity=dict(
+          description='An integer that controls the verbosity level, '
+                      '0 means no verbose output, increasing integers '
+                      'provide more verbosity.',
+          dataType='UInt32',
+          count=1,
+          constraints='',
+          defaultValue=0,
+          accessMode='ReadWrite'),
+
+      ),
+      commands=dict()
+    )
+
+    return ns
+
+
+  def __init__(self,
+               steps='1',
+               alpha=0.001,
+               clVerbosity=0,
+               implementation=None,
+               maxCategoryCount=None
+               ):
+
+    # Set default implementation
+    if implementation is None:
+      implementation = Configuration.get('nupic.opf.sdrClassifier.implementation')
+
+    # Convert the steps designation to a list
+    self.classifierImp = implementation
+    self.steps = steps
+    self.stepsList = eval("[%s]" % (steps))
```

Comment:
I don't understand why steps is a `str` in the first place, but previous discussions have resulted in standardizing on `str.format()` for substitution rather than `%`.